File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = []
    i = 2
    while i <= n:
        if n%i==0:
            if i not in answer:
                answer.append(i)
                n = n//i
            else:
                n = n//i
        else:
            i += 1
    return answer

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.stream.Collectors;

class Solution {
    public int[] solution(int n) {
        HashSet<Integer> primeFactors = new HashSet<>();
        int i = 2;
        while (n != 1) {
            if (n % i == 0) {
                primeFactors.add(i);
                n /= i;
            } else {
                i++;
            }
        }
        return primeFactors.stream().sorted().mapToInt(Integer::intValue).toArray();
    }
}